Userland/Libraries/LibWeb/HTML/WebSocket.cpp

```diff
@@ -215,8 +216,21 @@ void WebSocket::on_message(ByteBuffer message, bool is_text)
         dispatch_event(MessageEvent::create(EventNames::message, event_init));
         return;
     }
-    // type indicates that the data is Binary and binaryType is "blob"
-    // type indicates that the data is Binary and binaryType is "arraybuffer"
+    
+    if(m_binary_type == "blob") {
+        // type indicates that the data is Binary and binaryType is "blob"
+        TODO();
+    }else if(m_binary_type == "arraybuffer") {
+        // type indicates that the data is Binary and binaryType is "arraybuffer"
+        MessageEventInit event_init {};
```

Comment:
This looks a bit silly, how about just
```c++
MessageEventInit event_init {
    .data = JS::ArrayBuffer::...,
    .origin = url()
};
```

Reply:
Good idea, I changed the one I templated off of a few lines above as well, just to keep things consistent.